File: 0090-subsets-ii/0090-subsets-ii.cpp
class Solution {
public:
    void recursion(vector<int>& nums, int idx, vector<int>& temp, vector<vector<int>>& ans){
        if(idx >= nums.size()){
            if(find(begin(ans), end(ans), temp) == ans.end())
                ans.push_back(temp);
            return ;
        }
        
        temp.push_back(nums[idx]);
        recursion(nums, idx+1, temp, ans);
        temp.pop_back();
        recursion(nums, idx+1, temp, ans);
    }
    vector<vector<int>> subsetsWithDup(vector<int>& nums) {
        vector<vector<int>> ans;
        vector<int> temp;
        sort(begin(nums), end(nums));
        recursion(nums, 0, temp, ans);

        return ans;
    }
};

File: 0090-subsets-ii/0090-subsets-ii.java
class Solution {
    public void recursion(int[] nums, int idx, List<Integer> temp, List<List<Integer>> ans){
        if(idx >= nums.length){
            if(!ans.contains(temp))
                ans.add(new ArrayList<>(temp));
            return ;
        }
        temp.add(nums[idx]);
        recursion(nums, idx+1, temp, ans);
        temp.remove(temp.size()-1);
        recursion(nums, idx+1, temp, ans);
    }
    public List<List<Integer>> subsetsWithDup(int[] nums) {
        List<List<Integer>> ans = new ArrayList<>();
        List<Integer> temp = new ArrayList<>();
        Arrays.sort(nums);
        recursion(nums, 0, temp, ans);
        return ans;
    }
}